airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -353,7 +353,7 @@ data class GlobalCheckpoint(
                             AirbyteStateStats()
                                 .withRecordCount(destinationStats.recordCount.toDouble())
                     }
-                    it.additionalProperties.forEach { (key, value) ->
+                    additionalProperties.forEach { (key, value) ->
```

Comment:
Why that wasn't catch by tests? We should add a test for this (an Unit test). I am available to introduce mockk if needed.